File: sample/src/main/java/com/infullmobile/android/infullmvp/sample/activity/SampleActivity.java
package com.infullmobile.android.infullmvp.sample.activity;

import android.content.Context;
import android.content.Intent;
import com.infullmobile.android.infullmvp.InFullMvpActivity;
import com.infullmobile.android.infullmvp.sample.activity.di.SampleActivityGraph;
import javax.inject.Inject;

public class SampleActivity extends InFullMvpActivity<SampleActivityPresenter, SampleActivityView> {

    @Inject SampleActivityPresenter sampleActivityPresenter;
    @Inject SampleActivityView sampleActivityView;
    protected SampleActivityGraph sampleActivityGraph = new SampleActivityGraph(this);

    @Override
    public SampleActivityPresenter getPresenter() {
        return sampleActivityPresenter;
    }

    @Override
    public SampleActivityView getPresentedView() {
        return sampleActivityView;
    }

    @Override
    protected void injectIntoGraph() {
        sampleActivityGraph.inject(this);
    }

    public static Intent getIntent(Context context) {
        return new Intent(context, SampleActivity.class);
    }
}

File: sample-kotlin/src/main/java/com/infullmobile/android/infullmvp/sample/activity/SampleActivity.kt
package com.infullmobile.android.infullmvp.sample.activity

import android.content.Context
import android.content.Intent
import com.infullmobile.android.infullmvp.InFullMvpActivity
import com.infullmobile.android.infullmvp.sample.activity.di.SampleActivityGraph
import com.infullmobile.android.infullmvp.sample.activity.di.SampleActivityGraph.SampleActivityComponent
import com.infullmobile.android.infullmvp.sample.di_components.HasComponent
import javax.inject.Inject

class SampleActivity :
        InFullMvpActivity<SampleActivityPresenter, SampleActivityView>(),
        HasComponent<SampleActivityComponent> {

    @Inject lateinit var sampleActivityPresenter: SampleActivityPresenter
    @Inject lateinit var sampleActivityView: SampleActivityView
    val sampleActivityGraph = SampleActivityGraph(this)

    override fun injectIntoGraph() {
        sampleActivityGraph.inject(this)
    }

    override val presenter: SampleActivityPresenter
        get() = sampleActivityPresenter
    override val presentedView: SampleActivityView
        get() = sampleActivityView

    override val component: SampleActivityComponent
        get() = sampleActivityGraph.sampleActivityComponent

    companion object {
        fun getIntent(context: Context) = Intent(context, SampleActivity::class.java)
    }
}
